include/secp256k1.h

```diff
@@ -708,6 +708,74 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_pubkey_combine(
     size_t n
 ) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
+/** Opaque data structure that holds a parsed and valid "x-only" public key.
+ *  An x-only pubkey encodes a positive point. That is a point whose Y
+ *  coordinate is a quadratic residue. It is serialized using only its X
+ *  coordinate (32 bytes). A secp256k1_xonly_pubkey is also a secp256k1_pubkey
+ *  but the inverse is not true. Therefore, a secp256k1_pubkey must never be
+ *  interpreted as or copied into a secp256k1_xonly_pubkey.
+ *
+ *  The exact representation of data inside is implementation defined and not
+ *  guaranteed to be portable between different platforms or versions. It is
+ *  however guaranteed to be 64 bytes in size, and can be safely copied/moved.
+ *  If you need to convert to a format suitable for storage, transmission, or
+ *  comparison, use secp256k1_xonly_pubkey_serialize and
+ *  secp256k1_xonly_pubkey_parse.
+ */
+typedef struct {
+    unsigned char data[64];
+} secp256k1_xonly_pubkey;
```

Comment:
Would it make more sense to move this `typedef` up closer to the top of the file with all the other structures that are `typedef`ed?

Reply:
Hmm, seemed to be sensible to me to put this close to the functions related to xonly keys. The rest of the file is irrelevant to xonly. Other than being close to the other typedefs there doesn't seem to be a reason to do it with the clear downside of having to jump more when reading the xonly parts. But *shrugs*